File: ruby/series/series.rb
class Series
  def initialize(string)
    @string = string
  end

  def slices(num)
    num_slices = @string.length - num + 1
    raise ArgumentError if num_slices <= 0

    (0..(num_slices -1)).reduce([]) do |sum, i|
      sum << @string[i..(i + num - 1)]
    end
  end

end

File: python/series/series.py
def slices(series, length):
    if length not in range(1, len(series) + 1):
        raise ValueError("Invalid length")
    end_len = len(series) - length + 1
    return [list(map(int, series[i:i+length])) for i in range(end_len)]

    
